components/src/core/ToolbarMenu/ToolbarMenu.tsx

```diff
@@ -0,0 +1,281 @@
+import { DrawerNavGroup, NavItem } from '../Drawer';
+import makeStyles from '@material-ui/core/styles/makeStyles';
+import Menu, { MenuProps as standardMenuProps } from '@material-ui/core/Menu';
+import Typography from '@material-ui/core/Typography';
+import { Theme } from '@material-ui/core/styles/createMuiTheme';
+import useTheme from '@material-ui/core/styles/useTheme';
+import PropTypes from 'prop-types';
+
+export type ToolbarMenuClasses = {
+    root?: string;
+    dropdownArrow?: string;
+    icon?: string;
+    label?: string;
+    labelContent?: string;
+    menuItem?: string;
+};
+
+export type ToolbarMenuItem1 = Omit<NavItem, 'itemID'> & { itemID?: string };
+export type ToolbarMenuGroup1 = {
+    /** The color used for the text */
+    fontColor?: string;
+    /** The color used for icons */
+    iconColor?: string;
+    /** List of navigation items to render */
+    items: ToolbarMenuItem1[];
+    /** Text to display in the group header */
+    title?: string;
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        root: {
+            cursor: 'pointer',
+            color: (props: ToolbarMenuProps): string => props.color,
+        },
+        cursorPointer: {
+            cursor: 'pointer',
+        },
+        dropdownArrow: {
+            marginLeft: theme.spacing(0.5),
+        },
+        icon: {
+            marginRight: `${theme.spacing(1)}px`,
+            display: 'inline-flex',
+            fontSize: 'inherit',
+        },
+        label: {
+            textOverflow: 'ellipsis',
+            overflow: 'hidden',
+        },
+        labelContent: {
+            display: 'flex',
+        },
+        navGroups: {
+            '&:active, &:focus': {
+                outline: 'none',
+            },
+        },
+        rotateDropdownArrow: {
+            transform: 'rotate(180deg)',
+        },
+    })
+);
+
+export type ToolbarMenuProps = HTMLAttributes<HTMLDivElement> & {
+    /** The color used for the text elements
+     *
+     * Default: 'inherit'
+     */
+    color?: string;
+    /** A component to render for the icon */
+    icon?: JSX.Element;
+    /** Label Content */
+    label: ReactNode;
+    /** Custom content to be displayed in the menu */
+    menu?: JSX.Element;
+    /** Groups of menu items to display */
+    menuGroups?: ToolbarMenuGroup1[];
+    /** Property overrides for the MUI Menu */
+    MenuProps?: Omit<standardMenuProps, 'open'>;
+    /** Function called when the menu is opened */
+    onOpen?: () => void;
+    /** Function called when the menu is closed */
+    onClose?: () => void;
+    /** Custom classes for default style overrides */
+    classes?: Partial<ToolbarMenuClasses>;
+};
+
+const ToolbarMenuRenderer: React.ForwardRefRenderFunction<unknown, ToolbarMenuProps> = (
+    props: ToolbarMenuProps,
+    ref: any
+) => {
+    const {
+        classes = {},
+        /* eslint-disable @typescript-eslint/no-unused-vars */
```

Comment:
You need to re-enable this after the last expected unused variable. Right now, you've got that lint check turned off for the entire rest of the file.

If you want to disable it for a single line, use 'eslint-disable-next-line'